### PR TITLE
Added ClosureCommand

### PR DESCRIPTION
**Pull request by dktapps**
this is intended to replace PluginCommand and CommandExecutor, both of which are overengineered and unfit for purpose. Allowing a closure allows much greater flexibility.

We can't use this within the core yet, as plugins will expect PluginBase->getCommand() to return PluginCommand (with its associated setExecutor() and similar APIs). However, I think this is useful enough to add by itself.

### API changes
- Added a new `ClosureCommand` class. The callback accepted is similar to (and compatible with) the signatures required by `CommandExecutor` in the past, allowing easy migration.

### Behavioural changes
None. This class is not yet used in the core.

While I'd like to migrate directly to this, various plugins expect that `PluginCommand` will be returned by `PluginBase->getCommand()`.

In addition, there are questions about identifying plugin commands if this is used. This is no different a problem than we already see with regular `Command` users, but perhaps worth mentioning anyway.

## Backwards compatibility
No BC issues.

## Follow-up
- Deprecate and remove `PluginCommand` and `CommandExecutor`
- Perhaps #5960 ?

## Tests
Not directly tested, but this is trivially PHPStan-verifiable.

### CODE REVIEW

**Review by ShockedPlot7560 — DISMISSED** [review]

File: src/command/ClosureCommand.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+namespace pocketmine\command;
+
+use pocketmine\lang\Translatable;
+use pocketmine\utils\Utils;
+
+/**
+ * @phpstan-type Execute \Closure(CommandSender $sender, Command $command, string $commandLabel, list<string> $args) : mixed
+ */
+class ClosureCommand extends Command{
```
Comment (SOF3) [inline src/command/ClosureCommand.php]: why is this not final?

Comment (dktapps) [reply src/command/ClosureCommand.php]: I have no idea. My class templates are final by default ...

**Comment by dktapps** [comment]
I'm not sure whether we should allow replacing the closure handler a la `CommandExecutor->setExecutor()`. `setExecutor()` is used by some plugins to provide custom handlers for commands declared via `pocketmine.yml`. The command can just be re-registered with the new handler, so I'm not sure whether this is needed or not.

**Comment by SOF3** [comment]
Let's not overcomplicate the API. Mutable executor may as well be implemented within the closure as a delegation where it matters.

```php
$this->cmd = new ClosureCommand(xxx);
$this->cmd->setClosure(xxx);
register($this->cmd);
```

is barely different from

```php
$this->closure = xxx;
register(new ClosureCommand(fn(...$args) => ($this->closure)(...$args)));
```

**Comment by dktapps** [comment]
Delegate won't be available if the command was registered by the server from plugin.yml. That's basically the whole use case.

**Comment by dktapps** [comment]
setExecutor would become redundant if we decide to go forward with #6506

**Comment by dktapps** [comment]
I think this is fine as-is. If we decide to switch `plugin.yml` commands to this class in the future (as opposed to getting rid of them entirely), we can always add `setExecutor()` then when the time comes.

**Pull request merged**